tensorflow/contrib/gan/python/losses/python/losses_impl.py

```diff
@@ -306,6 +306,7 @@ def wasserstein_gradient_penalty(
     discriminator_scope,
     epsilon=1e-10,
     target=1.0,
+    mode='two_sided',
```

Comment:
Instead of adding a string that takes two values, what do you think about adding a boolean argument called something like 'one_sided' that defaults to 'false'?

Reply:
I decided to go with string arguments mainly because of two reasons:
1. To be somewhat consistent with the way `tf.layers` handles different modes. E.g. [`data_format` in `tf.layers.Conv2d`](https://www.tensorflow.org/api_docs/python/tf/layers/Conv2D).
2. If down the road there's an additional similar gradient penalty mode this solution would not need a breaking API change since the new mode could just be added to the string set.

That said, I'm happy to switch to a boolean argument if prefered.

Reply:
I don't have a strong opinion about this. I think the string arg is fine, and allows for future additions of new modes (is that likely?). @ebrevdo do you have anything to say?

Reply:
I'm pretty strongly in favor of a boolean, since I think there's a high likelihood that we won't need to bundle more features into this function, and if we do it might not be for a while. I prefer to think of the loss functions as atomic, which new functionality added via new functions (rather than modifying old ones). Your code is an obvious exception since it's such a small variant.

If you remain unconvinced, feel free to leave it. Otherwise, please switch the option here and the one in `train.py` to a boolean.

Thanks!

Reply:
👍 Switched to a boolean interface.